frame/evm/src/runner/stack.rs

```diff
@@ -1038,6 +1043,7 @@ where
 
 							let actual_size = Pallet::<T>::account_code_metadata(address).size;
 							if actual_size > pre_size {
+								fp_evm::set_storage_oog();
```

Comment:
I'm not quite understanding this PR; those hooks are at the EVM level. When it encounters an error like OutOfGas, it bubbles the error up to the top runner and reverts the changes that happened during this transaction. Why do we need this patch?